src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -206,6 +230,29 @@ public IPAddress(byte[] address)
             {
                 throw new ArgumentNullException(nameof(address));
             }
+
+            if (address.Length != IPAddressParserStatics.IPv4AddressBytes && address.Length != IPAddressParserStatics.IPv6AddressBytes)
+            {
+                throw new ArgumentException(SR.dns_bad_ip_address, nameof(address));
+            }
+
+            if (address.Length == IPAddressParserStatics.IPv4AddressBytes)
+            {
+                PrivateAddress = (uint)((address[3] << 24 | address[2] << 16 | address[1] << 8 | address[0]) & 0x0FFFFFFFF);
+            }
+            else
+            {
+                _numbers = new ushort[NumberOfLabels];
+
+                for (int i = 0; i < NumberOfLabels; i++)
+                {
+                    _numbers[i] = (ushort)(address[i * 2] * 256 + address[i * 2 + 1]);
+                }
+            }
+        }
+
+        public IPAddress(ReadOnlySpan<byte> address)
```

Comment:
Same question; is there a reason we can't implement the existing byte[]-based ctor in terms of the span-based one?